lib/dangermattic/plugins/milestone_checker.rb

```diff
@@ -35,35 +35,27 @@ class MilestoneChecker < Plugin
     # Checks if the pull request is assigned to a milestone.
     #
     # @return [void]
-    def check_milestone_set(fail_on_error: false)
+    def check_milestone_set(report_type: :warning)
       return unless milestone.nil?
 
       message = 'PR is not assigned to a milestone.'
-      sticky = false
-      if fail_on_error
-        failure(message, sticky: sticky)
-      else
-        warn(message, sticky: sticky)
-      end
+      reporter.report(message: message, type: report_type)
     end
 
     # Checks if the pull request's milestone is due to finish within a certain number of days.
     #
     # @param days_before_due [Integer] Number of days before the milestone due date for the check to apply (default: DEFAULT_DAYS_THRESHOLD).
-    # @param if_no_milestone [Symbol] Action to take if the pull request is not assigned to a milestone. Possible values:
-    #                        - :warn (default): Reports a warning.
-    #                        - :error: Reports an error.
-    #                        - :none or nil: Takes no action.
+    # @param report_type [Symbol] (optional) The type of report for the message. Types: :error, :warning (default), :message.
+    # @param report_type_if_no_milestone [Symbol] Action to take if the pull request is not assigned to a milestone. Possible values:
+    #                                 - :warning (default): Reports a warning.
+    #                                 - :error: Reports an error.
+    #                                 - :message: Reports a message.
+    #                                 - :none or nil: Takes no action.
```

Comment:
What do you think of using the same lean structure for both entries?

```suggestion
    # @param report_type [Symbol] (optional) The type of message for when the PR is has passed over the `days_before_due` threshold. Types: :error, :warning (default), :message.
    # @param report_type_if_no_milestone [Symbol] The type of message for when the PR is not assigned to a milestone. Types: :error, :warning (default), :message. You can also pass :none to not leave a message when there is no milestone.
```

Reply:
I had changed it slightly so it would be similar (before, the "if_no_milestone" would take just `:warn`). But I think the documentation was a bit off and mentioning `:none`. Thanks!